src/Compilers/CSharp/Portable/Lowering/ClosureConversion/ClosureConversion.cs

```diff
@@ -1205,12 +1206,17 @@ private BoundNode RewriteCatch(BoundCatchBlock node, ArrayBuilder<BoundExpressio
             else if (prologue.Count > 0)
```

Comment:
Are we guaranteed to have `prologue.Count > 0` when an `ExceptionFilterPrologueOpt` exists?

Reply:
No. Those are separate mechanisms for different things. `ExceptionFilterPrologueOpt` is used to evaluate expressions (such as the switch expression) that require statements as part of the evaluation. This prologue sets up the closure frame, e.g. converting a captured local into a field by creating the frame object for the variable.

---
In reply to: [456718544](https://github.com/dotnet/roslyn/pull/46054#discussion_r456718544) [](ancestors = 456718544)